lib/propolis/src/enlightenment/mod.rs

```diff
@@ -94,7 +95,7 @@ pub trait Enlightenment: Lifecycle + Send + Sync {
     /// CPUID leaves from 0x4000_0000 to 0x4000_00FF are reserved for the
     /// hypervisor's use. On entry, the caller must ensure that `cpuid` does not
     /// contain any leaf entries in this range.
-    fn add_cpuid(&self, cpuid: &mut CpuidSet) -> anyhow::Result<()>;
+    fn add_cpuid(&self, cpuid: &mut CpuidSet) -> Result<(), AddCpuidError>;
```

Comment:
take it or leave it: perhaps the `add_cpuid` method ought to still return an `anyhow::Result`, so that future implementations could return other errors as well as an `AddCpuidError`? i could maybe imagine an enlightenment stack which sets a more complicated set of hypervisor leaves wanting to return other errors if the enlightenment stack is misconfigured?

i dunno if this is actually something we want to do --- maybe it's better to constrain this to only return `AddCpuidError`s specifically to force other errors in the enlightenment stack's CPUID configuration to be returned elsewhere? what do you think?

Reply:
I'd be inclined to leave it for now.  Since it's largely an internal interface, we're free to revisit later if need be.

Reply:
I also sort of expect enlightenments not to fail this call for reasons other than a conflict (though my imagination might be limited). If a user asks to configure an enlightenment such that it has no valid CPUID values, I'd hope to catch that in the enlightenment's constructor. (So far so good with the Hyper-V prototype, anyway!)

Reply:
Yup, makes sense!